Dynavity/Dynavity/backlink/graph/Graph.swift

```diff
@@ -52,6 +52,25 @@ struct Graph<T: Hashable> {
         assert(checkRepresentation())
     }
 
+    /// Gets the given node if it exists.
+    /// If the node does not exists in the graph, return an empty optional.
+    func getNode(_ node: Node<T>) -> Node<T>? {
+        if !containsNode(node) {
+            return nil
+        }
+        return nodes.first { $0 == node }
+    }
+
+    /// Gets edge between source and destination node if it exists, otherwise return empty optional.
+    func getEdgeBetween(source: Node<T>, destination: Node<T>) -> Edge<T>? {
+        for edge in edges {
```

Comment:
Not sure if this is micro-optimizing, but I think it may be slightly more efficient to iterate from the edges in `adjList[source]` directly